R/monitor.R

```diff
@@ -0,0 +1,151 @@
+#' @param df_metrics A tidy dataframe of metrics over time, such as created by
+#' `vetiver_compute_metrics()`.
+#' @param metrics_pin_name Pin name for where the *metrics* are stored (as
+#' opposed to where the model object is stored with [vetiver_pin_write()]).
+#' @param initiate Are you starting a new set of monitoring metrics to
+#' pin/store? Defaults to `FALSE`.
+#' @param .estimate The variable in `df_metrics` containing the metric estimate.
+#' Defaults to `.estimate`.
+#' @param .metric The variable in `df_metrics` containing the metric type.
+#' Defaults to `.metric`.
+#' @param n The variable in `df_metrics` containing the number of observations
+#' used for estimating the metric.
+#'
+#' @return Both `vetiver_compute_metrics()` and `vetiver_pin_metrics()` return
+#' a dataframe of metrics. The `vetiver_plot_metrics()` function returns a
+#' `ggplot2` object.
+#'
+#' @details For arguments used more than once in your monitoring dashboard,
+#' such as `date_var`, consider using
+#' [R Markdown parameters](https://bookdown.org/yihui/rmarkdown/parameterized-reports.html)
+#' to reduce repetition and/or errors.
+#'
+#' @examples
+#' library(dplyr)
+#' library(parsnip)
+#' data(Chicago, package = "modeldata")
+#' Chicago <- Chicago %>% select(ridership, date, one_of(stations))
+#' training_data <- Chicago %>% filter(date < "2009-01-01")
+#' testing_data <- Chicago %>% filter(date >= "2009-01-01", date < "2011-01-01")
+#' monitoring <- Chicago %>% filter(date >= "2011-01-01", date < "2012-12-31")
+#' lm_fit <- linear_reg() %>% fit(ridership ~ ., data = training_data)
+#'
+#' library(pins)
+#' b <- board_temp()
+#'
+#' ## before starting monitoring, initiate the metrics and pin
+#' ## (for example, with the testing data):
+#' original_metrics <-
+#'     augment(lm_fit, new_data = testing_data) %>%
+#'     vetiver_compute_metrics(date, "week", ridership, .pred, .every = 4L) %>%
+#'     vetiver_pin_metrics(date, b, "lm_fit_metrics", initiate = TRUE)
+#'
+#' ## to continue monitoring with new data, compute metrics and update pin:
+#' new_metrics <-
+#'     augment(lm_fit, new_data = monitoring) %>%
+#'     vetiver_compute_metrics(date, "week", ridership, .pred, .every = 4L) %>%
+#'     vetiver_pin_metrics(date, b, "lm_fit_metrics")
+#'
+#' library(ggplot2)
+#' vetiver_plot_metrics(new_metrics, date) +
+#'     scale_size(range = c(2, 4))
+#'
+#' @export
+vetiver_compute_metrics <- function(data,
+                                    date_var,
+                                    .period,
+                                    truth, estimate, ...,
+                                    metric_set = yardstick::metrics,
+                                    .every = 1L,
+                                    .origin = NULL,
+                                    .before = 0L,
+                                    .after = 0L,
+                                    .complete = FALSE) {
+
+    rlang::check_installed("slider")
+    metrics_dots <- list2(...)
```

Comment:
Addressed in rewrite.

Capturing with `list2()` isn't going to work if you have class probability metrics like `roc_auc()`. These require that the user pass more bare column names through `...`, i.e. like `roc_auc(hpc_cv, truth = obs, VF, L, M)`.

Capturing with `list2()` will try and evaluate them immediately, which is too early. They need to be captured with `enquos()` OR you can pass the `...` all the way through to where you call `metric_set()`, which is what I ended up doing.